src/draw/lv_draw_img.c

```diff
@@ -193,16 +224,32 @@ void lv_draw_img_decoded(lv_draw_ctx_t * draw_ctx, const lv_draw_img_dsc_t * dsc
 /**********************
  *   STATIC FUNCTIONS
  **********************/
-
-LV_ATTRIBUTE_FAST_MEM static lv_res_t decode_and_draw(lv_draw_ctx_t * draw_ctx, const lv_draw_img_dsc_t * draw_dsc,
-                                                      const lv_area_t * coords, const void * src)
+LV_ATTRIBUTE_FAST_MEM static lv_res_t decode_and_draw(lv_draw_ctx_t * draw_ctx, lv_draw_img_dsc_t * draw_dsc,
+                                                      const lv_area_t * coords, const lv_img_src_t * src)
 {
     if(draw_dsc->opa <= LV_OPA_MIN) return LV_RES_OK;
 
-    _lv_img_cache_entry_t * cdsc = _lv_img_cache_open(src, draw_dsc->recolor, draw_dsc->frame_id);
+    lv_point_t size_hint;
+    size_hint.x =  lv_area_get_width(coords);
+    size_hint.y =  lv_area_get_height(coords);
 
+    lv_img_dec_dsc_in_t dsc = {.src = src, .color = draw_dsc->recolor, .size_hint = size_hint };
```

Comment:
Could we add opacity information to lv_img_dec_dsc_in_t (possibly make color uint32_t) as in #3287 ? It helps the recolor calculation a lot. Thanks!

Reply:
Sure!

Reply:
Please check my changes if it fits your expectations. I'm not sure how to test your change.

Reply:
Yes it does! As long as the opacity information is there and everything works as properly as before it's fine. Thank you!